docs/airframe-walkthrough.md

```diff
@@ -0,0 +1,1322 @@
+```scala
+package taskapp.server.service
+
+import taskapp.api.{Task, CreateTaskRequest, UpdateTaskRequest, TaskListResponse}
+import taskapp.server.repository.TaskRepository
+import wvlet.log.LogSupport
+
+class TaskService(repository: TaskRepository) extends LogSupport {
+  
+  def listTasks(): TaskListResponse = {
+    debug("Listing all tasks")
+    val tasks = repository.findAll()
+    TaskListResponse(tasks = tasks, total = tasks.size)
+  }
+  
+  def getTask(id: String): Option[Task] = {
+    debug(s"Getting task: ${id}")
+    repository.findById(id)
+  }
+  
+  def createTask(request: CreateTaskRequest): Task = {
+    info(s"Creating task: ${request.title}")
+    repository.create(request)
+  }
+  
+  def updateTask(id: String, request: UpdateTaskRequest): Option[Task] = {
+    info(s"Updating task: ${id}")
+    repository.update(id, request)
+  }
+  
+  def deleteTask(id: String): Boolean = {
+    info(s"Deleting task: ${id}")
+    repository.delete(id)
+  }
+}
+```
+
+**server/src/main/scala/taskapp/server/AppDesign.scala**
+```scala
+package taskapp.server
+
+import taskapp.server.repository.{TaskRepository, InMemoryTaskRepository}
+import taskapp.server.service.TaskService
+import wvlet.airframe.*
+
+object AppDesign {
+  
+  val design: Design = newDesign
+    // Bind configuration as singleton
+    .bind[AppConfig].toProvider { Config(env = "default").of[AppConfig] }
+    
+    // Bind repository interface to implementation
+    .bind[TaskRepository].to[InMemoryTaskRepository]
+    
+    // TaskService will automatically receive TaskRepository through constructor injection
+    .bind[TaskService].toSingleton
+}
+```
+
+**server/src/main/scala/taskapp/server/DIApp.scala**
+```scala
+package taskapp.server
+
+import taskapp.api.CreateTaskRequest
+import taskapp.server.service.TaskService
+import wvlet.log.{LogSupport, Logger}
+
+object DIApp extends App with LogSupport {
+  // Build and run the application using the design
+  AppDesign.design.build[TaskService] { taskService =>
+    info("Testing dependency injection")
+    
+    // Create some tasks
+    val task1 = taskService.createTask(CreateTaskRequest("Learn Airframe", "Complete the walkthrough"))
+    val task2 = taskService.createTask(CreateTaskRequest("Build app", "Create a task management app"))
+    
+    // List all tasks
+    val response = taskService.listTasks()
+    info(s"Total tasks: ${response.total}")
+    response.tasks.foreach { task =>
+      info(s"- ${task.title}: ${task.description}")
+    }
+  }
+}
+```
+
+### Key DI Features Demonstrated
+
+1. **Constructor Injection**: Dependencies are automatically injected through constructor parameters
+2. **Interface Binding**: Bind interfaces to implementations for easy testing and flexibility
+3. **Singleton Scoping**: Services can be singletons to share state
+4. **Provider Binding**: Complex object creation through provider functions
+
+## Step 3: Building REST APIs with airframe-http
```

Comment:
Let's focus on building an RPC service example by removing REST API definitions. Fix the example accordingly.